### PR TITLE
Create Rule “avoid-unnecessary-css-classes/rule”

### PR DESCRIPTION
**Pull request by Geordie88**
Automatically generated by Netlify CMS

### CODE REVIEW

**Review by bradystroud — DISMISSED** [review]

**Comment by tiagov8** [comment]
Hey @Geordie88 

I think I asked you to add me as the reviewer of this one... 

I have dismissed Brady approval since there are changes to be made

File: rules/avoid-unnecessary-css-classes/rule.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+type: rule
+title: Do You Avoid Unnecessary CSS Classes
```
Comment (tiagov8) [inline rules/avoid-unnecessary-css-classes/rule.md:3]: 1. We don't capitalize rules' titles, please fix
2. Missing "?" a the end

File: rules/avoid-unnecessary-css-classes/rule.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+---
+When making or editing CSS or HTML content it is important to avoid adding classes and ID's unnecessarily.
+
+ It can be tempting to add classes on elements, it is often the most obvious solution to CSS problems but doing so will lead to overly cluttered code and a host of overly specific solutions. When working on CSS it is almost always better to remove content rather than adding additional classes. 
+
+<!--endintro-->
+
+::: bad
+![Figure: Bad Example - The View-All-Link class unessessary as the arrow next class already exists](unessessary-class.jpg)
```
Comment (tiagov8) [inline rules/avoid-unnecessary-css-classes/rule.md:21]: 1. Please fix grammar

File: rules/avoid-unnecessary-css-classes/rule.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+:::
+
+::: good
+![Figure: Good Example - Instead the arrow next with an additional modifier was used](good-css-class.jpg)
```
Comment (tiagov8) [inline rules/avoid-unnecessary-css-classes/rule.md:25]: 1. Change from `arrow next` to `"arrow-next"` 
2. Check for other occurances

File: rules/avoid-unnecessary-css-classes/rule.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    noimage: true
+  - title: Tiago Araujo
+    url: https://www.ssw.com.au/people/tiago-araujo
+created: 2022-07-28T06:06:37.551Z
+guid: ca1b7770-298b-4035-b45d-a570fa0ee77b
+
+---
+When making or editing CSS or HTML content it is important to avoid adding classes and ID's unnecessarily.
+
+ It can be tempting to add classes on elements, it is often the most obvious solution to CSS problems but doing so will lead to overly cluttered code and a host of overly specific solutions. When working on CSS it is almost always better to remove content rather than adding additional classes.
```
Comment (tiagov8) [inline rules/avoid-unnecessary-css-classes/rule.md:16]: 1. Remove extra space at the beginning

File: rules/avoid-unnecessary-css-classes/rule.md [hunk 5]
```diff
@@ -13,14 +13,14 @@ guid: ca1b7770-298b-4035-b45d-a570fa0ee77b
...
 When making or editing CSS or HTML content it is important to avoid adding classes and ID's unnecessarily.
 
- It can be tempting to add classes on elements, it is often the most obvious solution to CSS problems but doing so will lead to overly cluttered code and a host of overly specific solutions. When working on CSS it is almost always better to remove content rather than adding additional classes. 
+It can be tempting to add classes on elements, it is often the most obvious solution to CSS problems but doing so will lead to overly cluttered code and a host of overly specific solutions. When working on CSS it is almost always better to remove content rather than adding additional classes. 
 
 <!--endintro-->
 
 ::: bad
-![Figure: Bad Example - The View-All-Link class unessessary as the arrow next class already exists](unessessary-class.jpg)
+![Figure: Bad Example - The View-All-Link class is unessessary, the arrow-next class already exists](unessessary-class.jpg)
```
Comment (tiagov8) [inline rules/avoid-unnecessary-css-classes/rule.md:21]: Hey @Geordie88 

1. Fix grammar mistake
2. Put all classes name in quotes, so the class name is not confused with the rest of the text   
   e.g. The "View-All-Link" class is... the "arrow-text" class is...

File: rules/avoid-unnecessary-css-classes/rule.md [hunk 6]
```diff
@@ -18,9 +18,9 @@ It can be tempting to add classes on elements, it is often the most obvious solu
 <!--endintro-->
 
 ::: bad
-![Figure: Bad Example - The View-All-Link class is unessessary, the arrow-next class already exists](unessessary-class.jpg)
+![Figure: Bad Example - The "View-All-Link" class is unessessary because the "arrow-next" class already exists](unessessary-class.jpg)
```
Comment (tiagov8) [inline rules/avoid-unnecessary-css-classes/rule.md:21]: 1. Fix "unecessary"

**Review by tiagov8 — CHANGES_REQUESTED** [review]
Hey @Geordie88 

as per https://youtu.be/mdoxkF0s0mo your good example is not correct.

1. Please fix by explaining you should reuse classes instead of creating new ones
**Tip:** Instead of images, you can use code in Markdown (easier to maintain)

PS. If you go further, the "arrow-next" class shouldn't have been created by whoever it was. So don't use it as an example

**Review by tiagov8 — APPROVED** [review]

**Pull request merged**